src/Renci.SshNet/Common/SshDataStream.cs

```diff
@@ -153,12 +143,21 @@ public void Write(string s, Encoding encoding)
         }
 
         /// <summary>
-        /// Reads a byte array from the SSH data stream.
+        /// Reads a length-prefixed byte array from the SSH data stream.
         /// </summary>
         /// <returns>
         /// The byte array read from the SSH data stream.
         /// </returns>
         public byte[] ReadBinary()
+        {
+            return ReadBinarySegment().ToArray();
+        }
+
+        /// <summary>
+        /// Reads a length-prefixed byte array from the SSH data stream,
+        /// returned as a view over the underlying buffer.
+        /// </summary>
+        internal ArraySegment<byte> ReadBinarySegment()
```

Comment:
later this will be used in `SshData` to remove reading out large arrays in e.g. `ChannelDataMessage`